frontend/src/components/modals/modalparts/CustomerModalParts.tsx

```diff
@@ -112,3 +122,104 @@ export const SelectRepresentatives: FC<{
     </div>
   </div>
 );
+
+export const AddOrModifyCustomer: FC<{
+  customers: RoadmapCreationCustomer[];
+  members: InviteRoadmapUser[];
+  initialCustomer?: RoadmapCreationCustomer;
+  error: boolean;
+  onSubmit: (customer: RoadmapCreationCustomer) => void;
+  onCancel: () => void;
+  onCloseError: () => void;
+}> = ({
+  customers,
+  members,
+  initialCustomer,
+  error,
+  onSubmit,
+  onCancel,
+  onCloseError,
+}) => {
+  const [colorType, setColorType] = useState(
+    initialCustomer ? 'pick' : 'generate',
+  );
+  const [customer, setCustomer] = useState(
+    initialCustomer || {
+      name: '',
+      email: '',
+      color: randomColor(customers),
+    },
+  );
+  const [representatives, setRepresentatives] = useState(
+    members
+      .filter(({ type }) => hasPermission(type, Permission.CustomerRepresent))
+      .map((member) => ({
+        ...member,
+        checked: initialCustomer
+          ? !!initialCustomer.representatives.find(
+              ({ email, checked }) => email === member.email && checked,
+            )
+          : false,
+      })),
+  );
+
+  const handleColorTypeChange = (value: string) => {
+    setColorType(value);
+    if (value === 'generate')
+      setCustomer({ ...customer, color: randomColor(customers) });
+  };
+
+  return (
+    <div className={classes(css.addOrEditMember)}>
+      <SelectCustomerInfo
+        name={customer.name}
+        email={customer.email}
+        colorType={colorType}
+        color={customer.color}
+        onNameChange={(name) => setCustomer({ ...customer, name })}
+        onEmailChange={(email) => setCustomer({ ...customer, email })}
+        onColorTypeChange={handleColorTypeChange}
+        onColorChange={(color) => setCustomer({ ...customer, color })}
+      />
+      {representatives.length > 0 && (
+        <SelectRepresentatives
+          representatives={representatives}
+          onRepresentativeChange={(idx, checked) => {
+            const copy = [...representatives];
+            copy[idx].checked = checked;
+            setRepresentatives(copy);
+          }}
+        />
+      )}
+      <Alert
+        show={error}
+        variant="danger"
+        dismissible
+        onClose={() => onCloseError()}
+      >
+        <Trans i18nKey="The combination of name and email should be unique" />
+      </Alert>
+      <div className={classes(css.buttons)}>
+        <button
+          className="button-small-filled"
+          type="button"
+          disabled={!customer.name || !customer.email}
+          onClick={() => onSubmit({ ...customer, representatives })}
```

Comment:
At the moment this allows submitting invalid fields. The first click is blocked and the field shows the error status, but the second click succeeds.

This and the member modal also have some other unexpected/undesirable interactions with the step form, for example <kbd>enter</kbd> submits the form, not the nested modal, advancing to the next step.

We should somehow change the submit action to submit the currently open edit modal instead, then the normal form validation will also catch the invalid fields.